src/dfx/src/commands/ledger/mod.rs

```diff
@@ -238,6 +242,34 @@ pub async fn notify_top_up(
     Ok(result)
 }
 
+pub async fn validate_subnet_type(agent: &Agent, subnet_type: &Option<String>) -> DfxResult<()> {
+    let result = agent
+        .update(
+            &MAINNET_CYCLE_MINTER_CANISTER_ID,
+            GET_SUBNET_TYPES_TO_SUBNETS_METHOD,
+        )
```

Comment:
This looks like it will blow up in some local development mode (unavoidable call to hard-coded ID that is only valid on mainnet, CMC is not necessarily even installed in local mode even if ledger is present)

Reply:
also, is it even necessary to validate? Doesn't the canister creation call fail if an invalid subnet type is supplied?

Reply:
That's a good point. I thought it might be nicer to do this validation on the dfx side as well. You're right though that canister creation will fail if an unknown subnet type is supplied (with a relevant error message). In local development mode, I don't think you ever create canisters with `dfx ledger` (you need to setup a ledger and CMC first correctly so I doubt users do that normally). That said, we can skip this validation, if you think it's not the best idea.

In this case, having a way to view all available types is even more useful I think (I was hoping that if you use the wrong one and then we show you the full list is helpful in order to figure out which subnet types you can use).

Reply:
I agree having a way to display available types is a good idea. And I think we can do all of what you suggest when handling errors, which will save us a an update call round-trip in most cases. (If the long-approved error codes were finally implemented this would be an even easier task. I have to start bothering people about that.)

Reply:
Ok, I can then remove the additional check through the update call and add an extra command to display types. Any recommendations on the level this command should be added at? E.g. `dfx canister ...` or just `dfx ...`?

Reply:
eeugh... hardest question so far 😄 I feel like this should be part of `canister create`, but maybe you could make it just a flag: `dfx canister create --available-subnet-types`. But I'm really not sure what the cleanest solution is. @ericswanson-dfinity do you have any good ideas?

Reply:
Since this is used with `dfx ledger create-canister`, what would you think about `dfx ledger subnet-types` ?  Technically it's call to the cycles minter, but the other `dfx ledger` operations are calls to the ledger and the cycles minter in combination.

Reply:
`subnet-types` or `show-subnet-types`? I really want to add autocomplete at some point, so the length wouldn't be _that_ big of an issue

Reply:
Either way, but a separate command for sure

Reply:
I have added a sub-command to list the subnet types and removed the automatic check when creating the canister as discussed. Let me know how it looks.